hazelcast/src/main/java/com/hazelcast/sql/impl/expression/datetime/Formatter.java

```diff
@@ -0,0 +1,791 @@
+                        if (digits == null || d < digits.length()) {
+                            parts.add(g.equals('G') ? symbols.getGroupingSeparator() : g.equals('D') ? (currency
+                                    ? symbols.getMonetaryDecimalSeparator() : symbols.getDecimalSeparator()) : g);
+                        } else if (fillMode) {
+                            f++;
+                        }
+                    } else if (g instanceof Integer) {
+                        if (digits == null) {
+                            StringBuilder r = new StringBuilder();
+                            for (int j = 0; j < (int) g; j++) {
+                                r.append('#');
+                            }
+                            parts.add(r);
+                        } else if (d < digits.length()) {
+                            int e = Math.min(d + (int) g, digits.length());
+                            StringBuilder r = new StringBuilder().append(digits, d, e);
+                            parts.add(pre ? r.reverse() : r);
+                            if (fillMode) {
+                                f += d + (int) g - e;
+                            }
+                        } else if (fillMode) {
+                            f += (int) g;
+                        }
+                        d += (int) g;
+                    }
+
+                    i++;
+                    if (f > 0 && (i == size() || (get(i) instanceof Anchorable && !((Anchorable) get(i)).isAnchored()))) {
+                        StringBuilder r = new StringBuilder();
+                        for (; f > 0; f--) {
+                            r.append(' ');
+                        }
+                        parts.add(r);
+                    }
+                }
+                if (pre) {
+                    Collections.reverse(parts);
+                }
+                parts.forEach(s::append);
+            }
+
+            @Override
+            public String toString() {
+                List<Object> mask = new ArrayList<>(this);
+                if (pre) {
+                    Collections.reverse(mask);
+                }
+                return mask.toString();
+            }
+        }
+
+        NumberFormat(String format) {
+            Matcher m = NUMERIC_TEMPLATE.matcher(format);
+            Mask mask = integerMask;
+            boolean currency = false, exponential = false, roman = false;
+            int i = 0, exponent = 0;
+
+            for (; m.find(); i = m.end()) {
+                if (m.start() > i) {
+                    mask.add(new Literal(format.substring(i, m.start())));
+                }
+                String group = m.group();
+                if (group.equals("FM")) {
+                    mask.toggleFillMode();
+                } else if (group.startsWith("V")) {
+                    exponent += group.length() - 1;
+                } else if (group.startsWith("F") || group.startsWith("\"")) {
```

Comment:
Where does the `F` come from? Do we need it? Aren't the quotes sufficient?

Also we don't support backslash as an escape. Quoting PG:
>If you want to have a double quote in the output you must precede it with a backslash, for example '\"YYYY Month\"'. Backslashes are not otherwise special outside of double-quoted strings. Within a double-quoted string, a backslash causes the next character to be taken literally, whatever it is (but this has no special effect unless the next character is a double quote or another backslash).